baseplate/integration/pyramid.py

```diff
@@ -43,9 +58,14 @@ def _on_new_request(self, event):
             # TODO: some metric for 404s would be good
             return
 
-        trace_id = request.headers.get("X-Trace", "no-trace")
-        parent_id = request.headers.get("X-Parent", "no-parent")
-        span_id = request.headers.get("X-Span", "no-span")
+        if self.trust_trace_headers:
+            trace_id = request.headers.get("X-Trace", "no-trace")
```

Comment:
Nothing's really using this yet, so we could definitely change them if it's an issue. I don't really feel strongly.